app/models/commercial.rb

```diff
@@ -1,10 +1,43 @@
 class Commercial < ActiveRecord::Base
+  require 'oembed'
+
   belongs_to :commercialable, polymorphic: true
 
-  attr_accessible :commercial_id, :commercial_type
+  attr_accessible :url
+
+  validates :url, presence: true
+  validates :url, format: URI::regexp(%w(http https))
+
+  def self.provider
```

Comment:
I think we can drop this anyway! This was necessary for the drop down menu. I guess we could also hard code the provider in the partial for backward compatibility!